physical/azure.go

```diff
@@ -0,0 +1,143 @@
+package physical
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/storage"
+	"io/ioutil"
+	"sort"
+	"strings"
+)
+
+// MaxBlobSize limits the value size per Blob to 4MB
+var MaxBlobSize = 1024 * 1024 * 4
+
+// AzureBackend is a backend storing data in Azure Blob Storage
+type AzureBackend struct {
+	container string
+	client    storage.BlobStorageClient
+}
+
+func newAzureBackend(conf map[string]string) (Backend, error) {
+	container, ok := conf["container"]
+
+	if !ok {
+		return nil, fmt.Errorf("Azure 'container' is required")
+	}
+
+	accountName, ok := conf["accountName"]
+
+	if !ok {
+		return nil, fmt.Errorf("An Azure 'accountName' is required")
+	}
+
+	accountKey, ok := conf["accountKey"]
+
+	if !ok {
+		return nil, fmt.Errorf("An Azure 'accountKey' is required")
+	}
+
+	client, err := storage.NewBasicClient(accountName, accountKey)
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create Azure client: %s", err)
+	}
+
+	client.GetBlobService().CreateContainerIfNotExists(container, storage.ContainerAccessTypePrivate)
+
+	backend := AzureBackend{container: container, client: client.GetBlobService()}
+	return backend, nil
+}
+
+//Delete removes the blob {key} from the container (recursively)
```

Comment:
Looking at the code in this function...is it actually recursive, or is the comment just out of date? I don't know the Azure API.